### PR TITLE
ECL-738: Show address on amend confirmation

### PR DESCRIPTION
**Pull request by paddynski-opencast**

### CODE REVIEW

File: app/uk/gov/hmrc/economiccrimelevyregistration/controllers/AmendmentRequestedController.scala [hunk 1]
```diff
@@ -44,7 +45,15 @@ class AmendmentRequestedController @Inject() (
       .get(request.session, request.internalId, SessionKeys.FirstContactEmailAddress)
       .map {
         case Some(firstContactEmailAddress) =>
-          Ok(view(firstContactEmailAddress, request.eclRegistrationReference))
+          val contactAddress =
+            Json.fromJson[EclAddress](Json.parse(request.session.get(SessionKeys.contactAddress).getOrElse(""))).asOpt
```
Comment (liandra-giacomine) [inline app/uk/gov/hmrc/economiccrimelevyregistration/controllers/AmendmentRequestedController.scala:49]: can we use the session service instead - it returns a none when it doesn't find the session key

**Review by liandra-giacomine — CHANGES_REQUESTED** [review]
could you add an integration test for this pls?

File: app/uk/gov/hmrc/economiccrimelevyregistration/controllers/AmendmentRequestedController.scala (hunk 1 above)
Comment (paddynski-opencast) [reply app/uk/gov/hmrc/economiccrimelevyregistration/controllers/AmendmentRequestedController.scala:49]: Tried that but I had a problem with the `Future`. Also we already use the session for other things on this view.

Comment (paddynski-opencast) [reply app/uk/gov/hmrc/economiccrimelevyregistration/controllers/AmendmentRequestedController.scala]: Made the change.

**Comment by paddynski-opencast** [comment]
Added tests.

**Review by liandra-giacomine — APPROVED** [review]

**Pull request merged**